gensim/models/base_any2vec.py

```diff
@@ -0,0 +1,781 @@
+                raise TypeError("You can't pass a generator as the sentences argument. Try an iterator.")
+            self.build_vocab(sentences, trim_rule=trim_rule)
+            self.train(
+                sentences, total_examples=self.corpus_count, epochs=self.epochs, start_alpha=self.alpha,
+                end_alpha=self.min_alpha, compute_loss=compute_loss)
+        else:
+            if trim_rule is not None:
+                logger.warning(
+                    "The rule, if given, is only used to prune vocabulary during build_vocab() "
+                    "and is not stored as part of the model. Model initialized without sentences. "
+                    "trim_rule provided, if any, will be ignored.")
+
+    # for backward compatibility (aliases pointing to corresponding variables in trainables, vocabulary)
+    @property
+    def iter(self):
+        return self.epochs
+
+    @property
+    def syn1(self):
+        return self.trainables.syn1
+
+    @syn1.setter
+    def syn1(self, value):
+        self.trainables.syn1 = value
+
+    @syn1.deleter
+    def syn1(self):
+        del self.trainables.syn1
+
+    @property
+    def syn1neg(self):
+        return self.trainables.syn1neg
+
+    @syn1neg.setter
+    def syn1neg(self, value):
+        self.trainables.syn1neg = value
+
+    @syn1neg.deleter
+    def syn1neg(self):
+        del self.trainables.syn1neg
+
+    @property
+    def syn0_lockf(self):
+        return self.trainables.vectors_lockf
+
+    @syn0_lockf.setter
+    def syn0_lockf(self, value):
+        self.trainables.vectors_lockf = value
+
+    @syn0_lockf.deleter
+    def syn0_lockf(self):
+        del self.trainables.vectors_lockf
+
+    @property
+    def layer1_size(self):
+        return self.trainables.layer1_size
+
+    @layer1_size.setter
+    def layer1_size(self, value):
+        self.trainables.layer1_size = value
+
+    @property
+    def hashfxn(self):
+        return self.trainables.hashfxn
+
+    @hashfxn.setter
+    def hashfxn(self, value):
+        self.trainables.hashfxn = value
+
+    @property
+    def sample(self):
+        return self.vocabulary.sample
+
+    @sample.setter
+    def sample(self, value):
+        self.vocabulary.sample = value
+
+    @property
+    def min_count(self):
+        return self.vocabulary.min_count
+
+    @min_count.setter
+    def min_count(self, value):
+        self.vocabulary.min_count = value
+
+    @property
+    def cum_table(self):
+        return self.vocabulary.cum_table
+
+    @cum_table.setter
+    def cum_table(self, value):
+        self.vocabulary.cum_table = value
+
+    @cum_table.deleter
+    def cum_table(self):
+        del self.vocabulary.cum_table
+
+    def __str__(self):
+        return "%s(vocab=%s, size=%s, alpha=%s)" % (
+            self.__class__.__name__, len(self.wv.index2word), self.vector_size, self.alpha
+        )
+
+    def build_vocab(self, sentences, update=False, progress_per=10000, **kwargs):
```

Comment:
I don't think `sentences` is the right term here.  `data_iterable` instead for consistency with the `train()` method?

Reply:
Is this a bad practice -- renaming method parameters in the subclass for better readability/clarity? (`data_iterable`->`documents` in `Doc2Vec` and `data_iterable`->`sentences` in `Word2Vec`).

Reply:
IMO, it would be best not to override the method but rather create another method. Maybe name the parent method `_build_vocab` and name the subclass method `build_vocab`. Then it's perfectly OK for the two methods to have different parameter names.